bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -720,8 +720,7 @@ public void asyncAddEntry(final byte[] data, final AddCallback cb,
      * @param ctx
      *            some control object
      */
-    public void asyncAddEntry(final long entryId, final byte[] data, final AddCallback cb, final Object ctx)
-            throws BKException {
+    public void asyncAddEntry(final long entryId, final byte[] data, final AddCallback cb, final Object ctx) {
```

Comment:
in LedgerHandleAdv, we should not allow adding entries without providing any entry id. Otherwise, the semantic is too complicated to maintain. As it is hard to tell, -1 is a user provided entry id or it is provided by calling addEntry without an entry id.

/cc @jvrao